ansible/roles/test/tasks/advanced-reboot.yml

```diff
@@ -106,15 +106,33 @@
         dest: /tmp/ports.json
       delegate_to: "{{ ptf_host }}"
 
+    - debug: msg="Defined new sonic image url is {{ new_sonic_image }}"
+      when: new_sonic_image is defined
+
+    - set_fact:
+        stay_in_target_image: false
+      when: stay_in_target_image is not defined
+
+    - set_fact:
+        cleanup_old_sonic_images: false
+      when: cleanup_old_sonic_images is not defined
+
     - block:
 
         - name: Save image version
           shell: 'sonic_installer list | grep Current | cut -f2 -d " "'
           register: current_sonic_image
           become: true
+          when: not stay_in_target_image|bool
```

Comment:
>|bool [](start = 40, length = 5)

not needed.

Reply:
It is needed when the parameter was passed in from ansible, the variable would be string.

Reply:
It is needed when the parameter was passed in from ansible, the variable would be string.

Reply:
we can convert it to bool in 'set_fact'

---
In reply to: [276479414](https://github.com/Azure/sonic-mgmt/pull/877#discussion_r276479414) [](ancestors = 276479414)

Reply:
we don't set fact if it is defined though

Reply:
addressed